### PR TITLE
Add atomic assignments and reads to logLevel

### PR DESCRIPTION
**Pull request by thebriz24**

### CODE REVIEW

File: log.go [hunk 1]
```diff
@@ -4,55 +4,91 @@ import (
...
+	case 3:
+		return "error"
+	case 4:
+		return "fatal"
+	default:
+		return "Log level error"
+	}
+}
+
+func getLogLevel() int {
```
Comment (brettallred) [inline log.go]: let's rename this to just logLevel()   

no get

Comment (brettallred) [inline log.go]: why getLogLeve() and RetriveLogLevel()

can we just make one method called LogLevel()

File: log.go [hunk 2]
```diff
@@ -4,55 +4,91 @@ import (
...
+// Fatal prints the format string with the prefix of "FATAL:"
 func Fatal(format string, v ...interface{}) {
-	if LogLevel <= 4 {
+	if getLogLevel() <= 4 {
 		log.Printf("FATAL:"+format, v...)
 	}
 }
+
+// ChangeLogLevel takes the logLevel string and changes the level in a thread-safe way
+func ChangeLogLevel(newLevel string) {
```
Comment (brettallred) [inline log.go]: SetLogLevelByName(level string)

Comment (brettallred) [inline log.go]: The counter to this would be SetLogLevel(level int)

File: log.go [hunk 3]
```diff
@@ -4,55 +4,92 @@ import (
...
-	if LogLevel <= 2 {
+	if atomic.LoadInt32(&logLevel) <= 2 {
 		log.Printf("WARN:"+format, v...)
 	}
 }
 
+// Error prints the format string with the prefix of "ERROR:"
 func Error(format string, v ...interface{}) {
-	if LogLevel <= 3 {
+	if atomic.LoadInt32(&logLevel) <= 3 {
```
Comment (brettallred) [inline log.go]: can we just call LogLevel() here?

Comment (thebriz24) [inline log.go]: Right now LogLevel() returns a string rather than an int. That can change if you'd like.

Comment (brettallred) [inline log.go]: Lets change that to LogLevelName() then

**Pull request merged**